tla-assignments/src/main/scala/at/forsyte/apalache/tla/assignments/Reordering.scala

```diff
@@ -0,0 +1,50 @@
+package at.forsyte.apalache.tla.assignments
+
+import at.forsyte.apalache.tla.lir.oper.TlaBoolOper
+import at.forsyte.apalache.tla.lir.{LetInEx, OperEx, TlaEx, TlaOperDecl}
+import at.forsyte.apalache.tla.lir.transformations.{TlaExTransformation, TransformationTracker}
+
+class Reordering[T]( ord : Ordering[T], rankingFn : TlaEx => T, tracker : TransformationTracker ) {
+
+  def reorder : TlaExTransformation = tracker.track {
+    // For now, only /\ and \/ are reordered, but the same process works for all commutative operators
+    case ex@OperEx( op@( TlaBoolOper.and | TlaBoolOper.or ), args@_* ) =>
+      val reorderedArgs = args.sortBy( rankingFn )( ord ) map reorder
+      if ( args == reorderedArgs ) ex
```

Comment:
What's the purpose of this check? iiuc, whenever `args == reorderedArgs` it will also be true that `ex == OperEx( op, reorderedArgs : _* )`. Right? So can skip the conditional here and on 18 and just return the constructed `OperEx` in the `else`?

The motive here (in addition to possibly removing unnecessary code) is the assumption that it's cheaper to just construct a new `OperEx` than to check all the arguments for equality.

Reply:
This is a canonical check, in practice, it is better to not introduce a new expression not for performance's sake, but because preserving the expression also preserves the UID. This matters, because a lot of auxiliary information is passed as a `UID -> X` map, so by keeping the expression whenever possible, you don't invalidate those maps.

Reply:
Ah! I see thanks for explaining :)

In that case, would it be valid to preserve the UUID even when we reorder the args? Or does that break the logic we need for the expression identification?

Reply:
I would advise against that, UIDs are, by definition, unique, and copying them over opens us up to a host of new, potentially very well hidden, bugs.